### PR TITLE
Add working directory flag to allow use of action from non-root directory

### LINKED ISSUE #18: How do you use this action in a folder not in the root directory?
Hello, 

I am new to working with Github actions. 

I am trying to run this action on a "user_interface" sub directory, can anyone explain to me how to do this?

Thanks!

### PR DESCRIPTION
**Pull request by LaurenceGA**
Fixes #18 

This action assumes that your `package.json` is in the root directory of your repository (or I guess that you have `eslint` globally installed). I think it's common to have eslint in your `package.lock` in a subdirectory. Hence, it's necessary to tell the action not only where to run `eslint`, but also where to find it. Unless I'm missing something, I'm pretty sure this is required for my use case anyway.

`eslint_flags` is fine for specifying _what to run eslint on_, but the action still looks for `eslint` in the root directory, so it's insufficient for this use case.

Please let me know if I missed anything!

- [x] Add test

### CODE REVIEW

**Comment by aslafy-z** [comment]
Reporting to Reviewdog will only match paths if eslint reports absolutes ones. I'm not able to test right now, is it the case?

**Comment by haya14busa** [comment]
Same as https://github.com/reviewdog/action-golangci-lint/pull/15, github-pr-check and github-check reporter doesn't support sub-directory run currently. https://github.com/reviewdog/reviewdog/issues/461

File: action.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ inputs:
   eslint_flags:
     description: "flags and args of eslint command. Default: '.'"
     default: '.'
+  working_directory:
```
Comment (haya14busa) [inline action.yml]: Can you change it to `workdir`?

**Comment by LaurenceGA** [comment]
> Reporting to Reviewdog will only match paths if eslint reports absolutes ones. I'm not able to test right now, is it the case?

It seems to me that `eslint` always reports absolute URLs anyway? So I assume there are other reasons that `reviewdog` can't be run from a subdirectory than the linter's output URLs 🤔

**Comment by LaurenceGA** [comment]
> Same as [reviewdog/action-golangci-lint#15](https://github.com/reviewdog/action-golangci-lint/pull/15), github-pr-check and github-check reporter doesn't support sub-directory run currently. [reviewdog/reviewdog#461](https://github.com/reviewdog/reviewdog/issues/461)

I've updated to try to allow for running `eslint` (and `reviewdog`) from the root directory, but on the given `workdir`.

It seems to have mostly worked. The check failed (as expected, I suppose) and the one with reporter=`github-pr-review` passed, but I guess it would have failed the actual review.

It was super easy to get `npm` to do its work in a subdirectory when run from root, but `eslint` was a bit trickier. 

A few things worth noting:
Even if you specify `workdir` to be a sub folder, `eslint`'s flags' file references are still relative to the root directory. So I made it that by default it passes "`workdir`" instead of "`.`", but this could be confusing if you try to override the flags.
So if you want to specify a subfolder of your subproject, you still have specify "`subproject`/`subfolder`/" instead of just "`subfolder`/", even if you have already specified `workdir`. So I doesn't entirely function like you'd expect a working directory to 😞
Also when I try running `eslint` on the subprojects' root directory from the main project's root directory, it seems to fail to ignore `node_modules`. Not sure why that is...

These may be reasons to just hold off until `reviewdog` supports [running in subdirectories](https://github.com/reviewdog/reviewdog/issues/461)...

File: entrypoint.sh [hunk 2]
```diff
@@ -4,19 +4,23 @@ cd "$GITHUB_WORKSPACE"
 
 export REVIEWDOG_GITHUB_API_TOKEN="${INPUT_GITHUB_TOKEN}"
 
-if [ ! -f "$(npm bin)/eslint" ]; then
-  npm install
+ESLINT_COMMAND="$(npm bin --prefix $INPUT_WORKDIR)/eslint"
```
Comment (haya14busa) [inline entrypoint.sh:7]: Can you just move to sub workdir instead?

```
cd "${GITHUB_WORKSPACE}/${INPUT_DIRECTORY}" || exit 1
```

reviewdog can support subdir run with the latest release. https://github.com/reviewdog/reviewdog/blob/master/CHANGELOG.md#v0100---2020-05-07

Comment (LaurenceGA) [reply entrypoint.sh]: Great work @haya14busa !
I'll update the PR and change the action argument to `directory` (?).

**Comment by LaurenceGA** [comment]
📝 Tests don't actually run properly as the PR is from my forked repo 🤔

**Comment by ChadTaljaardt** [comment]
Thanks for the fast response to the requested changed @LaurenceGA

**Comment by haya14busa** [comment]
The test worked with minor modifications. https://github.com/reviewdog/action-eslint/pull/25
As it's actually changing the workdir, we no longer need to prepend subdir in eslint_flags as you wrote (https://github.com/reviewdog/action-eslint/pull/19#issuecomment-582204391)

**Comment by haya14busa** [comment]
I merged it and released a new version.
I also changed the flag name back to workdir as I prefer workdir because I think it's more descriptive than just a directory.

Thank you for your contribution, @LaurenceGA !

**Pull request merged**